src/vscode-dts/vscode.proposed.chatProvider.d.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	export interface ChatResponseFragment {
+		index: number;
```

Comment:
Why is an index needed?

Reply:
It's the choice things. So, you have two choices and fragments like `(1, Hello), (1, World), (0, Salve), (1, !)` for  "Salve" and "Hello World!"

This needs some proper API, like a ResponseItem which emits events for its chunks